app/models/ability.rb

```diff
@@ -0,0 +1,11 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new
+    can :destroy, Post, author_id: user.id
+    can :destroy, Comment, author_id: user.id
+    can :destroy, Post, author_id: user.id if user.admin?
+    can :destroy, Comment, author_id: user.id if user.admin?
```

Comment:
- [ ] According to CanCanCan [docs](https://github.com/CanCanCommunity/cancancan/blob/develop/docs/define_check_abilities.md):
> By default, CanCanCan assumes no permissions: no one can do any action on any object.

This means that at the moment, your app only allows `:destroy` actions. Also, naming the abilities one by one seems kind of repetitive, remember to keep the code DRY. Consider doing something like the following:

```ruby
         if user.admin? 
           can :manage, :all
         else
           can :manage, Post, author: user
           can :manage, Comment, author: user
           can :read, :all
         end
```

Take this code as a reference to implement yours! 🤓 🔧 The `manage` method refers to performing all CRUD actions.

Reply:
fixed, thanks for the guidance 💯